### PR TITLE
Add network opts to pods

### PR DESCRIPTION
**Pull request by mheon**
Still needs tests, but this adds a large group of network options to `podman pod create`

### CODE REVIEW

**Comment by mheon** [comment]
List of things this resolves: 
#4718 
#4770 

Probably more

**Comment by mheon** [comment]
Also fixes #3837

File: cmd/podman/pod_create.go [hunk 1]
```diff
@@ -67,7 +55,7 @@ func init() {
 	flags.StringVarP(&podCreateCommand.Name, "name", "n", "", "Assign a name to the pod")
 	flags.StringVarP(&podCreateCommand.Hostname, "hostname", "", "", "Set a hostname to the pod")
 	flags.StringVar(&podCreateCommand.PodIDFile, "pod-id-file", "", "Write the pod ID to the file")
-	flags.StringSliceVarP(&podCreateCommand.Publish, "publish", "p", []string{}, "Publish a container's port, or a range of ports, to the host (default [])")
+	//flags.StringSliceVarP(&podCreateCommand.Publish, "publish", "p", []string{}, "Publish a container's port, or a range of ports, to the host (default [])")
```
Comment (TomSweeneyRedHat) [inline cmd/podman/pod_create.go]: Why is this commented out?

**Comment by TomSweeneyRedHat** [comment]
One questions, otherwise LGTM

File: cmd/podman/pod_create.go (hunk 1 above)
Comment (mheon) [reply cmd/podman/pod_create.go]: Because I forgot to remove it :smile:

**Comment by rhatdan** [comment]
LGTM

**Comment by TomSweeneyRedHat** [comment]
docs test is barfing.

**Comment by mheon** [comment]
Argh. Needs manpages and bash completions. Will tackle tomorrow.

**Comment by mheon** [comment]
I appear to have broken Varlink, as well. Will fix.

**Comment by mheon** [comment]
Also fixes #2808

**Comment by mheon** [comment]
Added manpages. Next bash completions, fix varlink, add tests

**Comment by mheon** [comment]
Now also fixes https://github.com/containers/libpod/issues/4432

**Comment by mheon** [comment]
I think I got the Varlink bug. Now moving on to testing.

**Comment by mheon** [comment]
Starting to add tests

**Comment by mheon** [comment]
Tests mostly done. Missing: `--ip`, `--mac-address`, `--network`.

**Comment by mheon** [comment]
There are some thorns with `--net=host` the way we're doing it right now. I think that we're close enough, though. Pushing the last few tests now, I think we're good to merge once things are green.

**Comment by mheon** [comment]
@rhatdan @vrothberg @giuseppe @baude @TomSweeneyRedHat PTAL

**Comment by rhatdan** [comment]
LGTM
I like all of the tests.

**Comment by mheon** [comment]
Test failure: `/etc/hosts` appears to have unpredictable amounts of whitespace between IP and host for hosts we add, which is messing with my comparison. Will investigate further tomorrow.

**Comment by mheon** [comment]
Tests should start going green now, last issue is fixed

**Comment by mheon** [comment]
Everything is green - PTAL and merge

**Pull request merged**